### PR TITLE
Initial support for optogenetic stimulation

### PR DESCRIPTION
**Pull request by iraikov**
This pull request adds support for optogenetic stimulation, based on PyRHO code.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -24,19 +24,9 @@ networkx = "^2.8.4"
...
-pydata-sphinx-theme = {version = "0.9.0", optional = true, extras = ["docs"]}
-readthedocs-sphinx-search = {version = "^0.1.2", optional = true, extras = ["docs"]}
-sphinx-autodoc-typehints = {version = "^1.18.1", optional = true, extras = ["docs"]}
-myst-parser = {version = "^0.17.2", optional = true, extras = ["docs"]}
-numpydoc = {version = "^1.3.1", optional = true, extras = ["docs"]}
-sphinx-togglebutton = {version = "^0.3.1", optional = true, extras = ["docs"]}
-sphinx-copybutton = {version = "^0.5.0", optional = true, extras = ["docs"]}
-myst-nb = {version = "^0.15.0", optional = true, extras = ["docs"]}
-sphinx-click = {version = "^4.3.0", optional = true, extras = ["docs"]}
-sphinxcontrib-mermaid = {version = "^0.7.1", optional = true, extras = ["docs"]}
```
Comment (skim0119) [inline pyproject.toml:37]: Can we revert this block of code? We need it for documentation build in ReadTheDocs.

File: README.md [hunk 2]
```diff
@@ -1,22 +1,122 @@
 <div align='center'>
 <h1>MiV - Simulator</h1>
-
-[![Documentation Status][badge-documentation]][link-documentation]
 </div>
 
 Bio-physical neural network simulator for Mind-in-Vitro in context of processing and computation.
 
-## Documentation
+## Generic installation
```
Comment (skim0119) [inline README.md:7]: We moved the installation and user guidelines into the documentation. It looks like the file is copied from the previous branch. Can we revert the change?

File: pyproject.toml (hunk 1 above)
Comment (iraikov) [reply pyproject.toml:37]: Yes, I will try to add it back. Poetry was getting stuck computing some sphinx-related dependencies, but I will try adding them back and regenerating the lock file.

File: README.md (hunk 2 above)
Comment (iraikov) [reply README.md:7]: Yes, absolutely

File: pyproject.toml (hunk 1 above)
Comment (iraikov) [reply pyproject.toml:37]: > Can we revert this block of code? We need it for documentation build in ReadTheDocs.

Actually, why are the documentation tools dependencies? Is there no separate category for documentation dependencies? Thanks for clarifying.

Comment (skim0119) [reply pyproject.toml:37]: Yes, that was the purpose for `extras = ["docs"]`. Although, I'll double check the poetry, I think they changed the syntax a little for optional dependencies.

Comment (skim0119) [reply pyproject.toml:37]: I checked the `poetry` documentation for any update, and it looks like we still need to specify dependencies here with `optional=true` tag, and group them in `[tool.poetry.extras]` section (we already did below). Looks like `extras=["docs"]` tag is not needed anymore.

Comment (iraikov) [reply pyproject.toml:37]: Ok, thank you for checking. I am going to add them back.

Comment (iraikov) [reply pyproject.toml:37]: Unfortunately, poetry gets stuck resolving dependencies when I add this block back. I am think I am going to try adding them one by one. What is the poetry command line to add dependencies with optional=true?

Comment (frthjf) [reply pyproject.toml:37]: I think `poetry add ... -E docs -G dev` [(reference)](https://python-poetry.org/docs/cli/#options-4)

Comment (iraikov) [reply pyproject.toml:37]: Thanks @frthjf I ended up using poetry add --optional and it turned out there is a conflict between myst and particular versions of sphinx, so I added version constraints that don't send poetry into tailspin. @skim0119 I would appreciate if you take a look and see if the doc can build ok.

Comment (skim0119) [reply pyproject.toml:37]: You can check the build here:
https://miv-simulator.readthedocs.io/en/feature-pyrho/

**Comment by skim0119** [comment]
Looks all good to me
